src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperDescriptorConventions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Host
+{
+    /// <summary>
+    /// A library of methods used to generate <see cref="TagHelperDescriptor"/>s for view components.
+    /// </summary>
+    public static class ViewComponentTagHelperDescriptorConventions
+    {
+        /// <summary>
+        /// The key in a <see cref="TagHelperDescriptor"/> property bag containing 
+        /// the short name of a view component.
+        /// </summary>
+        public static readonly string ViewComponentNameKey = "ViewComponentName";
+
+        /// <summary>
+        /// The key in a <see cref="TagHelperDescriptor"/> property bag containing
+        /// a custom type name for a view component's tag helper representation.
+        /// </summary>
+        public static readonly string ViewComponentTagHelperNameKey = "ViewComponentTagHelperName";
+
+        /// <summary>
+        /// Each custom type name for a view component's tag helper representation will begin with this header.
+        /// </summary>
+        public static readonly string ViewComponentTagHelperNameHeader = "__Generated__";
```

Comment:
I only see one reference to this property. Does it need to be on this class or can it be inlined?